docs/unpackaged.rst

```diff
@@ -1,14 +1,161 @@
+
+The task ``deploy_pre``, which is part of the ``dependencies`` flow, is responsible for deploying these bundles.
+
+Metadata that's not intended to be delivered to all installations of the product should *not* be included in ``unpackaged/pre``.
+
+``unpackaged/post``: Configuration After Package Install
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Projects often include metadata that is genuinely part of the application, but either cannot be delivered as part of a managed package or for operational reasons should not be. For example, a product that wishes to deliver ``TopicsForObjects`` metadata cannot do so as part of a managed package, because that type of metadata is not packageable.
+
+``unpackaged/post`` is the home for metadata of this kind, which is stored in subdirectories such as ``unpackaged/post/first``. CumulusCI's out-of-the-box flows that build an org, such as ``dev_org`` and ``install_prod``, always deploy metadata bundles found in ``unpackaged/post``, making it a full-fledged part of the application. Further, it's easy to include ``unpackaged/post`` metadata in customer-facing installers run via MetaDeploy.
+
+The task ``deploy_post``, which is part of the ``config_dev``, ``config_qa``, and ``config_managed`` flows, is responsible for deploying these bundles.
+
+Metadata that's not intended to be delivered to all installations of the product should *not* be included in ``unpackaged/post``. It's critical for managed package projects that this metadata include namespace tokens (see :ref:`namespace-injection`).
+
+``unpackaged/config``: Tailoring Orgs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Projects may come with more than one supported configuration in their CumulusCI automation. For example, projects may support distinct, tailored ``dev_org``, ``qa_org``, and ``install_prod`` flows, each of which performs different setup for their specific use case. Unpackaged metadata stored in ``unpackaged/config`` is a tool to support operational needs to tailor orgs to different configurations. 
+
+For example, a testing-oriented scratch org may wish to deploy a customized set of Page Layouts to help testers easily visualize data under test. Such page layouts could be stored in ``unpackaged/config/qa``.
+
+
+Unpackaged Metadata Folder Structure
+------------------------------------
+
+All unpackaged metadata is stored in the ``unpackaged`` directory tree. Within this directory are the three top-level directories, ``unpackaged/pre``, ``unpackaged/post``, and ``unpackaged/config``.
+
+These trees contain metadata bundles in Metadata API format: that is, a directory containing a ``package.xml`` manifest and Metadata API-format source code. CumulusCI does not support Salesforce DX format for unpackaged bundles.
```

Comment:
Actually since the merge of the 2gp branch, I believe if you point `deploy` at a folder that doesn't have a package.xml, it will use sfdx to convert it in a tempdir to deploy. But this may not work well with namespace tokens in some locations if they get in the way of the conversion. There is potential to make it work with a scheme like this:
1. Copy sfdx format to a tempdir.
2. Do namespace injection.
3. Convert to mdapi.